react/tips/common-mistakes-in-react/react-dependency-array-for-useeffect-ii.md

```diff
@@ -0,0 +1,82 @@
+---
+author: kapnobatai136
+  
+aspects:
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: tip
+
+links:
+  - '[Use deep compare effect](https://github.com/kentcdodds/use-deep-compare-effect){website}'
+
+---
+
+# Dependency array for useEffect (Part 2)
+
+---
+## Content
+
+The first consists of moving the dependency array outside of the component, although in some cases this might not be possible.

```

Comment:
```suggestion
Using the `useEffect()` hook with a dependency array might lead to some unexpected bugs. This is due to how the comparison is made. Instead of the value being compared, React compares the references. In this insight, we'll take a look at how to avoid any issues. 

The first consists of moving the dependency array outside of the component, although in some cases this might not be possible.
```